File: DevTools/RemoteDebugInterfaces/src/main/java/info/smart_tools/smartactors/remote_debug/models/ioc/IocValue.java
package info.smart_tools.smartactors.remote_debug.models.ioc;

import java.util.List;

public class IocValue {

    private final String key;
    private final IocStrategy[] strategy;

    public IocValue(final String key, final List<IocStrategy> strategy) {
        this.key = key;
        this.strategy = strategy.toArray(new IocStrategy[0]);
    }

    public String getKey() {
        return key;
    }

    public IocStrategy[] getStrategy() {
        return strategy;
    }
}

File: DevTools/SmartActorsIdeaPlugin/src/main/kotlin/info/smart_tools/smartactors/devtools/ioc_viewer/tool/IocValue.kt
package info.smart_tools.smartactors.devtools.ioc_viewer.tool

data class IocModule(val name: String, val version: String)

data class IocStrategy(val module: IocModule, val strategy: Any)

data class IocValue(val key: String, val strategies: List<IocStrategy>)
